File: EdgePartition/CLUGP/src/ustc.ddl/utility/FileConvert.java
package ustc.ddl.utility;

import ustc.ddl.properties.GlobalConfig;

import java.io.*;
import java.sql.SQLOutput;
import java.util.Collection;
import java.util.Hashtable;
import java.util.Properties;

public class FileConvert {
    public static void main(String[] args) throws IOException {
        Hashtable<Integer, Integer> ht = new Hashtable<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream("newpartition/src/data/698.edges")));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(new FileOutputStream("newpartition/src/data/D698.edges")));
        String line;
        int lineNum = 0;
        int id = 0;
        while((line = br.readLine()) != null){
            if (line.startsWith("#")) continue;
            String[] edgeValues = line.split(" ");
            int srcVid = Integer.parseInt(edgeValues[0]);
            int endVid = Integer.parseInt(edgeValues[1]);
            int newSrcVid;
            int newEndVid;
            if(ht.containsKey(srcVid)) {
                newSrcVid = ht.get(srcVid);
            } else {
                newSrcVid = id++;
                ht.put(srcVid, newSrcVid);
            }
            if(ht.containsKey(endVid)) {
                newEndVid = ht.get(endVid);
            } else {
                newEndVid = id++;
                ht.put(endVid, newEndVid);
            }
            if (lineNum != 0)  {
                bw.write("\n");
                bw.flush();
            }
            bw.write(newSrcVid + "\t" + newEndVid);
            bw.flush();
            lineNum++;
        }
        System.out.println("Number of Nodes: " + id);
        System.out.println("Number of Edges: " + lineNum);
        br.close();
        bw.close();
    }
}

File: EdgePartition/CLUGP/C++/FileConvert.cpp
#include "FileConvert.h"
//#include "globalConfig.h"

void convertFile(const std::string& inputFilePath, const std::string& outputFilePath) {
	std::unordered_map<int, int> ht;
	std::ifstream inputFile(inputFilePath);
	std::ofstream outputFile(outputFilePath);
	std::string line;
	int lineNum = 0;
	int id = 0;

	while (std::getline(inputFile, line)) {
		if (line[0] == '#') continue;
		std::istringstream iss(line);
		int srcVid, endVid;
		iss >> srcVid >> endVid;
		int newSrcVid, newEndVid;

		// Check for srcVid
		auto srcIt = ht.find(srcVid);
		if (srcIt != ht.end()) {
			newSrcVid = srcIt->second;
		} else {
			newSrcVid = id++;
			ht[srcVid] = newSrcVid;
		}

		// Check for endVid
		auto endIt = ht.find(endVid);
		if (endIt != ht.end()) {
			newEndVid = endIt->second;
		} else {
			newEndVid = id++;
			ht[endVid] = newEndVid;
		}
		ht.clear();
		if (lineNum != 0) {
			outputFile << "\n";
		}
		outputFile << newSrcVid << "\t" << newEndVid;
		lineNum++;
	}

	std::cout << "Number of Nodes: " << id << std::endl;
	std::cout << "Number of Edges: " << lineNum << std::endl;

	inputFile.close();
	outputFile.close();
}
